docs/node/migrations/overview.mdx

```diff
@@ -0,0 +1,26 @@
+---
+sidebar_position: 1
+sidebar_label: "Overview"
+id: migrations-overview
+title: Migrations
+description: Overview of Kwil's migration system
+slug: /node/migrations/overview
+---
+
+This section provides an overview of the migration system in Kwil. Migrations are used to upgrade the network to a new version or to a new network with minimal downtime. Depending on the nature of the upgrade, migrations can be coordinated or uncoordinated.
+
+## Rolling upgrades
+
+Whenever network has to be upgraded to a newer versions of `kwild` binaries which doesn't have any breaking changes to the consensus rules, the upgrade can be done in a rolling fashion. In a rolling upgrade, the validators can upgrade their nodes one by one without stopping the network and still ensure that the nodes agree on the app hash for the block.
```

Comment:
> If a new a version of `kwild` has no breaking changes to the consensus rules, a network upgrade can be done in a rolling fashion. In a rolling upgrade, the validators can upgrade their nodes one by one without stopping the network and still ensure that the nodes agree on the application state at each block. This avoids creating a new network, which would have a new genesis block and chain ID.  When possible, Kwil will supports rolling upgrades. However, between major releases, it may not be practical to support rolling upgrades, thus necessitating a migration process as described in the next section.